base/include/ArchiveSpaceManager.h

```diff
@@ -0,0 +1,71 @@
+#pragma once
+#include "Module.h"
+#include "AIPExceptions.h"
+
+class ArchiveSpaceManagerProps : public ModuleProps
+{
+public:
+	ArchiveSpaceManagerProps(uint32_t _lowerWaterMark, uint32_t _upperWaterMark, string _pathToWatch, int _samplingFreq)
+	{
+		lowerWaterMark = _lowerWaterMark;
+		upperWaterMark = _upperWaterMark;
+		pathToWatch = _pathToWatch; 
+		samplingFreq = _samplingFreq;
+		
+		auto totalSpace = boost::filesystem::space(pathToWatch);
+		if ((lowerWaterMark > upperWaterMark) || (upperWaterMark > totalSpace.available))
```

Comment:
upperWaterMark can be greater than totalSpaceAvailable! this check should be done on the capacity of the disk, the same space function provides capacity, free and available